yt_dlp/extractor/tiktok.py

```diff
@@ -911,3 +911,47 @@ def _real_extract(self, url):
         if self.suitable(new_url):  # Prevent infinite loop in case redirect fails
             raise UnsupportedError(new_url)
         return self.url_result(new_url)
+
+
+class TikTokLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\.-]+)/live'
+    IE_NAME = 'tiktok:live'
+
+    def _real_extract(self, url):
+        uploader_id = self._match_id(url)
+        webpage = self._download_webpage(url, uploader_id, headers={'User-Agent': 'User-Agent:Mozilla/5.0'})
+        room_id = self._html_search_regex(r'snssdk\d*://live\?room_id=(\d+)', webpage, 'room ID', default=None)
+        if room_id:
+            live_detail_url = f'https://www.tiktok.com/api/live/detail/?aid=1988&roomID={room_id}'
+            # live_detail_url from https://github.com/Pauloo27/tiktok-live/blob/master/index.js#L23
+            video_js_data = self._download_json(live_detail_url, room_id)
+            title = (traverse_obj(video_js_data, ('LiveRoomInfo', 'title'), expected_type=str)
+                     or self._html_search_meta(['og:title', 'twitter:title'], webpage, fatal=True))
+            # thumbnail = traverse_obj(video_js_data, ('LiveRoomInfo', 'coverUrl'))
```

Comment:
you can inline these in the return (do include `thumbnail`)

IMO title shouldn't be fatal, you could change `fatal=True` to `default=''`

Reply:
Because `coverUrl` will show as `fail to get resource`, I haven't found the other live preview cover link yet.

Reply:
In addition to `uniqueId`, the user has `id` and `nickname`, but the user option seems to be only `uploader_id: Nickname or id of the video uploader.`
Like the example in description.

Reply:
if you want to extract those other user values and maintain consistency with the other TikTok IEs:
`uploader_id` = `id`
`creator` = `nickname`

and for `thumbnail` you could this:
```py
thumbnail = traverse_obj(video_js_data, ('LiveRoomInfo', 'coverUrl'), expected_type=url_or_none)
```
> you can inline these in the return

Reply:
I found that replacing `720x720` of `coverUrl` with `gcp` can get the `avatar`, maybe because there is no thumbnail, if the URL has `gcp` then it is a thumbnail. (No more examples confirmed, most of them are `720x720`.)